File: app/src/main/java/com/kamaltatyana/yandextestapp/adapter/ImageAdapter.java
package com.kamaltatyana.yandextestapp.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.kamaltatyana.yandextestapp.R;
import com.kamaltatyana.yandextestapp.pojo.Images;

import java.util.ArrayList;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.MyViewHolder> {

    private ArrayList<Images> mImagesArrayList;
    private Context mContext;
    private OnItemClickListener mListener;

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        mListener = listener;
    }

    public ImageAdapter(Context context, ArrayList<Images> imagesArrayList) {
        mContext = context;
        mImagesArrayList = imagesArrayList;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View inflater = LayoutInflater.from(mContext).inflate(R.layout.item_rv_images, parent, false);
        return new MyViewHolder(inflater);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {

        Images images = mImagesArrayList.get(position);

        String imageUrl = images.getmImageUrl();

        /**
         *  Получени изображений с Glide
         **/
        Glide.with(holder.itemView.getContext())
                .load(imageUrl)
                .dontAnimate()
                .error(R.drawable.images_error)
                .centerCrop()
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(holder.mImageView);
    }


    @Override
    public int getItemCount() {
        return mImagesArrayList.size();
    }


    public class MyViewHolder extends RecyclerView.ViewHolder {

        public ImageView mImageView;

        public MyViewHolder(View itemView) {
            super(itemView);
            mImageView = itemView.findViewById(R.id.image_view);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (mListener != null) {
                       int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            mListener.onItemClick(position);
                        }
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/kamaltatyana/redgallery/adapter/ImageAdapter.kt
package com.kamaltatyana.redgallery.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.kamaltatyana.redgallery.R
import com.kamaltatyana.redgallery.adapter.ImageAdapter.MyViewHolder
import com.kamaltatyana.redgallery.pojo.Images
import java.util.*

class ImageAdapter(private val mContext: Context, private val mImagesArrayList: ArrayList<Images>) : RecyclerView.Adapter<MyViewHolder>() {
    private var mListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    fun setOnItemClickListener(listener: OnItemClickListener?) {
        mListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val inflater = LayoutInflater.from(mContext).inflate(R.layout.item_rv_images, parent, false)
        return MyViewHolder(inflater)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val images = mImagesArrayList[position]
        val imageUrl = images.getmImageUrl()
        /**
         * Получени изображений с Glide
         */
        Glide.with(holder.itemView.context)
                .load(imageUrl)
                //.dontAnimate()
                //.error(R.drawable.images_error)
                //.centerCrop()
                //.diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(holder.mImageView)
    }

    override fun getItemCount(): Int {
        return mImagesArrayList.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mImageView: ImageView

        init {
            mImageView = itemView.findViewById(R.id.image_view)
            itemView.setOnClickListener {
                if (mListener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        mListener!!.onItemClick(position)
                    }
                }
            }
        }
    }

}